packages/cfd/src/Containers/__tests__/cfd-password-modal.spec.js

```diff
@@ -0,0 +1,513 @@
+            disableCFDPasswordModal: mockFn,
+        };
+
+        render(
+            <Router history={history}>
+                <CFDPasswordModal {...props} />
+            </Router>
+        );
+
+        fireEvent.click(await screen.findByRole('button', { name: /Submit proof/i }));
+
+        await waitFor(() => {
+            expect(mockFn).toHaveBeenCalledWith(false);
+            expect(mockDisableCFDPasswordModalFn).toHaveBeenCalled();
+            expect(mockFn).toHaveBeenCalledWith(false);
+        });
+    });
+
+    it('should show success message with options to transfer now or later when password has been updated successfully', async () => {
+        const props = {
+            ...mock_props,
+            is_cfd_password_modal_enabled: true,
+            is_cfd_success_dialog_enabled: true,
+            is_password_modal_exited: true,
+            account_type: { category: 'real', type: 'financial_stp' },
+            getAccountStatus: mockFn,
+            platform: 'mt5',
+            is_eu: true,
+            is_fully_authenticated: true,
+        };
+
+        render(
+            <Router history={history}>
+                <CFDPasswordModal {...props} />
+            </Router>
+        );
+
+        expect(await screen.findByRole('button', { name: /Maybe later/i }));
+        expect(await screen.findByRole('button', { name: /Transfer now/i }));
+    });
+
+    it('should display IcMt5SyntheticPlatform icon in Success Dialog', async () => {
+        const props = {
+            ...mock_props,
+            is_cfd_password_modal_enabled: true,
+            is_cfd_success_dialog_enabled: true,
+            account_status: { status: ['mt5_password_not_set', 'dxtrade_password_not_set'] },
+            getAccountStatus: mockFn,
+            platform: 'mt5',
+            error_type: 'PasswordError',
+            is_eu: true,
+            account_type: { category: 'real', type: 'synthetic' },
+        };
+        render(
+            <Router history={history}>
+                <CFDPasswordModal {...props} />
+            </Router>
+        );
+
+        expect(await screen.findByTestId('IcMt5SyntheticPlatform')).toBeInTheDocument();
+    });
+
+    it('should display IcMt5FinancialPlatform icon in Success Dialog', async () => {
+        const props = {
+            ...mock_props,
+            is_cfd_password_modal_enabled: true,
+            is_cfd_success_dialog_enabled: true,
+            account_status: { status: ['mt5_password_not_set', 'dxtrade_password_not_set'] },
+            getAccountStatus: mockFn,
+            platform: 'mt5',
+            error_type: 'PasswordError',
+            account_type: { category: 'real', type: 'financial' },
+        };
+        render(
+            <Router history={history}>
+                <CFDPasswordModal {...props} />
+            </Router>
+        );
+
+        expect(await screen.findByTestId('IcMt5FinancialPlatform')).toBeInTheDocument();
```

Comment:
Is it possible to mock the `Icon` component  instead of adding `data-testid` for each:

`
jest.mock('@deriv/components', () => {
    const original_module = jest.requireActual('@deriv/components');

    return {
        ...original_module,
        Icon: jest.fn((props) => <div>{props.icon}</div>),
    };
});`

Reply:
Will make the change